apps/oidc-provider-nest/src/main.ts

```diff
@@ -20,18 +29,81 @@ async function bootstrap() {
     cors: true
   });
 
-  const clientsService = app.select(ClientMetadataModule).get(ClientMetadataService, { strict: true });
-
-  const clients = await clientsService.loadClientMetadaForOidcSetup();
-
-  OpenIdProvider.build(clients);
-
-  enableOIDCDebug(OpenIdProvider.provider);
+  // Sets up various parameters used during deployment
+  // When passing in multiple args in the monorepo you need to use this format:
+  // --args='-h=6','-w=9','-d=3'
+  const { argv } = yargs(process.argv.slice(2))
+    .scriptName('oidc-provider-nest')
+    .option('s', {
+      alias: 'setup',
+      description: 'Create default admin user and oidc-angular client metadata',
+      type: 'boolean',
+      implies: ['n', 't', 'r', 'e', 'p']
+    })
+    .option('x', {
+      alias: 'dropSchema',
+      description: "Sets the TypeORM 'dropSchema' value to true",
+      type: 'boolean'
+    })
+    .option('n', {
+      alias: 'clientName',
+      description: 'With -s, determines what to set the default client_name as',
+      type: 'string',
+      nargs: 1
+    })
+    .option('t', {
+      alias: 'clientSecret',
+      description: 'With -s, determines what to set the default client_secret as',
+      type: 'string',
+      nargs: 1
+    })
+    .option('r', {
+      alias: 'redirectUri',
+      description: 'With -s, determines what to set the default client redirect_uri as',
+      type: 'string',
+      nargs: 1
+    })
+    .option('e', {
+      alias: 'admin-email',
+      description: 'Email used for the admin account',
+      type: 'string',
+      nargs: 1
+    })
+    .option('p', {
+      alias: 'admin-password',
+      description: 'Password used for the admin account',
+      type: 'string',
+      nargs: 1
+    })
+    .help()
+    .alias('help', 'h');
+
+  if (argv.x) {
+    console.warn('dropSchema set to true! Dropping all data in db');
+  }
+  if (argv.setup) {
+    console.log('Setup detected; creating defaults...');
+    OpenIdProvider.build();
+    console.warn("After setup, you will need to restart without '-s' so the oidc-admin client can be loaded");
+  } else {
+    console.log('Not creating defaults... Loading clients');
+    const service = app.select(ClientMetadataModule).get(ClientMetadataService, { strict: true });
+
+    const clients = await service.loadClientMetadaForOidcSetup();
+    console.log('Clients:', clients);
+    OpenIdProvider.build(clients);
+  }
+
+  // Only needed during development
+  // enableOIDCDebug(OpenIdProvider.provider);
```

Comment:
Can use `environment.production` from the environments file. That way this isn't a manual comment in/out to enable it which would could also mark them as changes for future commits if one forgets to comment them out for production.